packages/block-editor/src/components/block-list/use-block-props/use-block-refs.js

```diff
@@ -18,6 +18,50 @@ import { BlockRefs } from '../../provider/block-refs-provider';
 /** @typedef {import('@wordpress/element').RefCallback} RefCallback */
 /** @typedef {import('@wordpress/element').RefObject} RefObject */
 
+function addToMap( map, id, value ) {
+	let setForId = map.get( id );
+	if ( ! setForId ) {
+		setForId = new Set();
+		map.set( id, setForId );
+	}
+	setForId.add( value );
+}
```

Comment:
Any reason to not reuse the same logic from #60879?

Reply:
The situation is very similar, but not entirely identical. Here, when adding refs to the map, we are merely registering them, without firing listeners. It's only when the ref callback actually sets the `ref.current` when the listeners are fired.

Maybe it's possible to merge the two implementations, but I'd rather do it in steps.